rfd/0191-workload-id-config-ux.md

```diff
@@ -0,0 +1,1454 @@
+return true for the WorkloadIdentity to be issued.
+
+If the `deny` ruleset is non-empty, then all rules within the set must return
+false for the WorkloadIdentity to be issued.
+
+Each rule consists of either a set of conditions or an expression. A rule 
+cannot have both conditions and an expression.
+
+When conditions have been provided within a rule, then all conditions within the
+rule must return true for the rule to return true. A condition consists of a
+single target attribute, a single operator and a single expected value.
+
+Initially, the following operators will be supported:
+
+- `equals`: The target attribute must equal the given value. 
+- `not_equals`: The target attribute must not equal the given value 
+- `matches`: The target attribute must be a string and must match the
+  given regex pattern.
+- `not_matches`: The target attribute must be a string and must not match
+  the given regex pattern.
+- `in`: The target attribute must equal one of the values in the given list.
+- `not_in`: The target attribute must not equal any of the values in the given
+  list.
+
+Future iterations may introduce additional operators as we better understand the
+types of conditions that will be useful to users.
+
+Alternatively, a rule may consist of a single expression. This expression is
+configured by the user in the predicate language with access to the same
+attributes as the conditions and templating. This expression must return a 
+boolean value. For example:
+
+```yaml
+kind: workload_identity
+version: v1
+metadata:
+  name: my-workload-identity
+spec:
+  spiffe:
+    # Implicitly, this WorkloadIdentity can only be issued to requester's
+    # with a successful join via the GitLab join method.
+    id: /gitlab/{{ join.gitlab.project_path }}/{{ join.gitlab.environment }}
+  rules:
+    allow:
+    - expression: join.gitlab.environment == "production"
+```
+
+See "Predicate Language" for further details.
+
+### Attributes
+
+The attributes available for templating and rule evaluation will come from
+three sources:
+
+- `join`: From the attestation performed when the bot joined. This will be 
+  specific to the join method used. E.g `join.gitlab.project_path`.
+- `workload`: From the workload attestation performed by `tbot`. The presence of 
+  these attributes is dependent on whether workload attestation has been
+  performed by `tbot` and which form of attestation has been performed. E.g
+  `workload.k8s.namespace`.
+- `user`: From the identity of the user calling the WorkloadIdentity API. E.g
+  `user.bot_name` or `user.traits`. This information does not directly come from
+  attestation but may be useful for referencing administrative or organizational
+  values.
+
+Attributes take a hierarchical form which can be navigated within templates and
+rules using `.`. For example, `join.gitlab.project_path`.
+
+Documentation will be generated with an exhaustive list of available attributes
```

Comment:
I think these are kind of covered in the protos below, but it'd be helpful to have the initial planned list of attributes just covered in yaml or a table or something here, somewhere in this doc. Mainly in a more easily parseable format than proto definitions. Not a hard requirement from me, but would be appreciated. (Unless it is and I'm just missing it somewhere.)

Reply:
It's a little beyond what I can put into this RFD without exploding the size - the Protos just for the join attributes come out to about 300 lines (https://github.com/gravitational/teleport/blob/67529d2facfaac6d67c18dfe8dceeb49339784f3/api/proto/teleport/workloadidentity/v1/attributes.proto). Represented as a table or similar, this would be even longer 😅

Reply:
Fair enough.